src/Features/Core/Portable/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -29,10 +31,24 @@ public async Task<(ISymbol?, TextSpan)> GetSymbolAndBoundSpanAsync(Document docu
 
             if (symbol is null)
             {
-                return (null, semanticInfo.Span);
+                return (null, project, semanticInfo.Span);
             }
 
-            return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
+            // If this document is not in the primary workspace, we may want to search for results
+            // in a solution different from the one we started in. Use the starting workspace's
+            // ISymbolMappingService to get a context for searching in the proper solution.
+            var mappingService = services.GetRequiredService<ISymbolMappingService>();
+            var mapping = await mappingService.MapSymbolAsync(document, symbol, cancellationToken).ConfigureAwait(false);
```

Comment:
This bit of code, including the comment, is the important change, and is copied directly from our Find All Refs code :)